tests/regression/whole_axis_access.cpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2019, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#include <gridtools/common/integral_constant.hpp>
+#include <gridtools/sid/rename_dimension.hpp>
+#include <gridtools/stencil/cartesian.hpp>
+#include <gridtools/stencil/positional.hpp>
+
+#include <stencil_select.hpp>
+#include <test_environment.hpp>
+
+namespace {
+    using namespace gridtools;
+    using namespace stencil;
+    using namespace cartesian;
+
+    struct functor {
+        using out = inout_accessor<0>;
+        using in = in_accessor<1, extent<>, 4>;
+        using k_pos = in_accessor<2>;
+        using param_list = make_param_list<out, in, k_pos>;
+
+        template <class Eval>
+        GT_FUNCTION static void apply(Eval &&eval) {
+            auto &&res = eval(out());
```

Comment:
Performance-wise it would most probably be better to have `res` being a non-reference type and just assign to `out` in the last step (`eval(out()) = res;`). This makes sure that the compiler does compile not generate accesses to main memory for `res` in each loop iteration (it _might_ also be able to optimize them away otherwise as we make use of `restrict`, but no guarantee I guess).